python/kwiver/sprokit/tests/sprokit/adapters/test-adapter_data_set.py

```diff
@@ -0,0 +1,272 @@
+    add_get_helper(ads, ads.add_timestamp, ads.get_port_data_timestamp, kvt.Timestamp(), "timestamp")
+    add_get_helper(ads, ads.add_f2f_homography, ads.get_port_data_f2f_homography, kvt.F2FHomography(1), "f2f_homography")
+
+# Now test overwriting
+def test_overwrite():
+    from kwiver.vital import types as kvt
+    from kwiver.sprokit.adapters import adapter_data_set
+    from kwiver.sprokit.pipeline import datum
+
+    OVERWRITE_PORT = "test_overwrite_port"
+    ads = adapter_data_set.AdapterDataSet.create()
+
+    # Overwriting with same datum
+    ads.add_datum(OVERWRITE_PORT, datum.new("d2"))
+    overwrite_helper(ads.add_datum, ads.get_port_data, datum.new("d3"), "datum_string", OVERWRITE_PORT)
+
+    # Overwriting with completely different types
+    overwrite_helper(ads.add_datum, ads.get_port_data, datum.new(12), "datum_int", OVERWRITE_PORT)
+    overwrite_helper(ads.add_string_vector, ads.get_port_data_string_vector, datum.VectorString(["baz", "qux"]), "string_vector", OVERWRITE_PORT)
+    overwrite_helper(ads.add_timestamp, ads.get_port_data_timestamp, kvt.Timestamp(100, 10), "timestamp", OVERWRITE_PORT)
+    overwrite_helper(ads.add_value, ads.get_port_data, 15, "int", OVERWRITE_PORT)
+    overwrite_helper(ads.add_double_vector, ads.get_port_data_double_vector, datum.VectorDouble([4, 8]), "double_vector", OVERWRITE_PORT)
+
+def test_iter():
+    from kwiver.vital import types as kvt
+    from kwiver.sprokit.adapters import adapter_data_set
+    from kwiver.sprokit.pipeline import datum
+
+    ads = adapter_data_set.AdapterDataSet.create()
+
+    # Construct a few elements
+    string_value = "string_value"
+    timestamp_value = kvt.Timestamp(1000000000, 10)
+    vector_string_value = datum.VectorString(["element1", "element2"])
+
+    # Add those elements
+    ads.add_string("string_port", string_value)
+    ads.add_timestamp("timestamp_port", timestamp_value)
+    ads.add_string_vector("vector_string_port", vector_string_value)
+
+    for port, dat in ads:
+        if port == "string_port":
+            if dat.get_string() != string_value:
+                test_error("Didn't retrieve correct string value on first iteration")
+        elif port == "timestamp_port":
+            if dat.get_timestamp() != timestamp_value:
+                test_error("Didn't retrieve correct timestamp value on second iteration")
+        elif port == "vector_string_port":
+            if dat.get_string_vector() != vector_string_value:
+                test_error("Didn't retrieve correct string vector on third iteration")
+        else:
+            test_error("unknown port: {}".format(port))
+
+
+if __name__ == "__main__":
+    import os
+    import sys
+
+    if not len(sys.argv) == 4:
+        test_error("Expected three arguments")
+        sys.exit(1)
+
+    testname = sys.argv[1]
+
+    os.chdir(sys.argv[2])
+
+    sys.path.append(sys.argv[3])
+
+    from kwiver.sprokit.util.test import *
```

Comment:
I have a few issues this block: 

1. IMHO `import *` is bad practice. It makes it confusing where things are coming from and it prevents the use of certain static analysis tools. Either `from kwiver.sprokit.util import test as <foobar>` and use `<foobar>.name` or do `from kwiver.sprokit.util.test import (<the>, <members>, <you>, <want>)`. There are rare cases where `import *` is acceptable, but I don't think this is one of them. 

2. This import should not be in the `if __name__ == "__main__":` block. If you were to import the functions from this module and try to call them, they would Error when they reached a name like `test_error`. Is there a problem with putting it at the top of the module?

3. Can we get rid of the `os.chdir`, and `sys.path.append`? They are awful fragile constructs. Usually when I see these patterns there is another more robust way to implement the same functionality.

Reply:
To answer point number 3, it turns out that both of those calls are unnecessary. I'll create a branch to fix that for all of sprokit

Reply:
This should be handled by #1056

Reply:
Resolved in 90e7c1907abf246285692646feaa83e2e645df20